add cartographer package

cirocosta opened the pull request:
## What this PR does / why we need it

This PR adds the [cartographer](https://github.com/vmware-tanzu/cartographer) package as an addon having the Package bits coming out of [`vmware-tanzu/package-for-cartographer`](https://github.com/vmware-tanzu/package-for-cartographer). 

~I'm currently marking this PR as a draft as there are at least two details to finish up:~

~1. [x] `package-for-cartographer` being made public~
~2. [x] image being hosted under `projects.registry.vmware.com`~


## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Add Cartographer v0.2.2 package
```

## Which issue(s) this PR fixes

closes #3024 

## Describe testing done for PR

0. create a cluster provisioned with kapp-controller and cert-manager
1. submit the `metadata.yaml` and `0.2.2/package.yaml` files
2. install the package using the tanzu cli

```bash
CARTOGRAPHER_VERSION=0.2.2
tanzu package install cartographer \
     --package-name cartographer.community.tanzu.vmware.com \
     --version $CARTOGRAPHER_VERSION
```
```console
| Installing package 'cartographer.community.tanzu.vmware.com'
| Getting package metadata for 'cartographer.community.tanzu.vmware.com'
| Creating service account 'cartographer-default-sa'
| Creating cluster admin role 'cartographer-default-cluster-role'
| Creating cluster role binding 'cartographer-default-cluster-rolebinding'
| Creating package resource
/ Waiting for 'PackageInstall' reconciliation for 'cartographer'
/ 'PackageInstall' resource install status: Reconciling
 Added installed package 'cartographer'
```

3. run the smoke test from `package-for-cartographer` (see https://github.com/vmware-tanzu/package-for-cartographer/blob/c8db5d70b0ad231cddbeb0633f002b88439e7aac/tests/01-test-basic.sh)


## Special notes for your reviewer

~As I mentioned before, the repository is currently private, so, `ghcr.io` will require credentials for pulling the image. Fortunately `secretgen-controller` helps here:~

```
tanzu secret registry add ghcr --username <gh_user> --password <gh_password> --server ghcr.io --export-to-all-namespaces
```

~before `tanzu package install` will do the job (or .. just wait for the repository to be made public _soon^{tm}_)~

Commit:
add cartographer package

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>

cirocosta commented on addons/packages/cartographer/0.2.2/package.yaml:
```diff
@@ -0,0 +1,24 @@
+    spec:
+      fetch:
+      - imgpkgBundle:
+          image: ghcr.io/vmware-tanzu/package-for-cartographer@sha256:5a58e3d9760b7972f0168161625cfd7be01fb55aff0b0983739e2d39f6794c1e
```
should be `projects.registry.vmware.com`

cirocosta commented on addons/packages/cartographer/0.2.2/README.md:
```diff
@@ -0,0 +1,120 @@
+# Package for Cartographer
```
this is straight from the `vmware-tanzu/package-for-cartographer` repository where the contents have been written really focusing on that repository itself and not `community-edition` - is there a specific way that we should be writing this?

cirocosta replied:
went ahead with mimicking harbor and kpack

cirocosta commented on addons/packages/cartographer/0.2.2/README.md:
```diff
@@ -0,0 +1,120 @@
+
+- [kapp-controller]
+- [Tanzu CLI]
+- [cert-manager]
```
> _It seems cartographer has some dependencies on other packages being available in the cluster (I could spot at least cert-manager) It would be ideal to have this pre-requisite documented_

cirocosta replied:
similarly to `harbor`, added a section to the installation where `tanzu package install` for cert-manager is performed before `cartographer`

cirocosta commented on addons/packages/cartographer/0.2.2/README.md:
````diff
@@ -0,0 +1,120 @@
+```bash
+tanzu package install cartographer \
+  --package-name cartographer.community.tanzu.vmware.com \
+  --version 0.0.0
````
feedback on slack:

> _There’s no values that can be customized, it’ll be ideal to also note this fact_

same as the other _todo_, it'd be good to check how other packages report configuration values (or lack thereof) and improve this

cirocosta replied:
added a section that mentions that there are no values to be configured.

jorgemoralespou commented:
@cirocosta could you add to the longDescription in PackageMetadata that cert-manager is required? It's currently the only place this dependency can be noted for the end users without reading the docs.

Commit:
cartographer: mention cert-manager in metadata

the requirement of having cert-manager installed is currently the only
discoverable by looking at the docs - having it under `PackageMetadata`
allows one to discover that dependency by simply looking up the
description about the package.

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>

cirocosta commented:
@jorgemoralespou just did it in 642cdcc2af0152f98518e0ad3b0d7d20052a785c 

```diff
diff --git a/addons/packages/cartographer/metadata.yaml b/addons/packages/cartographer/metadata.yaml
index 8f159a55..44e4b540 100644
--- a/addons/packages/cartographer/metadata.yaml
+++ b/addons/packages/cartographer/metadata.yaml
@@ -13,7 +13,7 @@ spec:
     Cartographer is a Kubernetes native Choreographer. It allows users to
     configure K8s resources into re-usable Supply Chains that can be used to
     define all of the stages that an Application Workload must go through to
-    get to an environment.
+    get to an environment. Requires cert-manager to be installed.
   maintainers:
   - name: VMware
```

wdyt? feel free to provide a suggestion 😁

jorgemoralespou commented:
LGTM.
